src/main/java/org/broadinstitute/hellbender/utils/haplotype/EventMap.java

```diff
@@ -126,20 +134,38 @@ protected void processCigarForInitialEvents() {
                 case EQ:
                 case X:
                 {
-                    for( int iii = 0; iii < elementLength; iii++ ) {
-                        final byte refByte = ref[refPos];
-                        final byte altByte = alignment[alignmentPos];
-                        if( refByte != altByte ) { // SNP!
-                            if( BaseUtils.isRegularBase(refByte) && BaseUtils.isRegularBase(altByte) ) {
-                                final List<Allele> snpAlleles = new ArrayList<>();
-                                snpAlleles.add( Allele.create( refByte, true ) );
-                                snpAlleles.add( Allele.create( altByte, false ) );
-                                proposedEvents.add(new VariantContextBuilder(sourceNameToAdd, refLoc.getContig(), refLoc.getStart() + refPos, refLoc.getStart() + refPos, snpAlleles).make());
-                            }
+                    final List<Integer> mismatchBlockStarts = new ArrayList<>();    //inclusive
```

Comment:
This does the correct thing.  The new code makes a MNP out of any haplotype with a MNP.  It will not make a MNP out of two different haplotypes with adjacent SNPs.

In general we produce far more haplotypes than actually exist due to kmerization.  For example, when k = 10 in our assembly and you have two SNPs 15 bases apart we will produce four different haplotypes for the 2 x 2 = 4 independent combinations.  This is not an issue with adjacent SNPs because kmers span them.

But let's suppose that for unknown reasons there's a hole in this argument and that we produce a haplotype with two adjacent SNPs falsely in phase.  In that case, every read will contradict the haplotype and so the MNP allele will have a very low LOD or QUAL, as the case may be.

Reply:
Do we need to worry about that last case?  Is there a way we can test it?

Would this generate multiallelics when the adjacent SNPs are not in phase?

Reply:
It should be easy enough to make an integration test.

For the same reasons as above I wouldn't expect multiallelics, but an integration test will cover this, too.  I just need to dig up or create a small bam with unphased SNPs.

Reply:
Thanks!